File: superset-frontend/cypress-base/cypress/e2e/dashboard_list/filter.test.ts
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import { DASHBOARD_LIST } from 'cypress/utils/urls';
import { setGridMode, clearAllInputs } from 'cypress/utils';
import { setFilter } from '../dashboard/utils';

describe('Dashboards filters', () => {
  before(() => {
    cy.visit(DASHBOARD_LIST);
    setGridMode('card');
  });

  beforeEach(() => {
    clearAllInputs();
  });

  it('should allow filtering by "Owner" correctly', () => {
    setFilter('Owner', 'alpha user');
    setFilter('Owner', 'admin user');
  });

  it('should allow filtering by "Modified by" correctly', () => {
    setFilter('Modified by', 'alpha user');
    setFilter('Modified by', 'admin user');
  });

  it('should allow filtering by "Status" correctly', () => {
    setFilter('Status', 'Published');
    setFilter('Status', 'Draft');
  });
});

File: tests/unit_tests/dashboard_folders/filter_test.py
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.
"""Tests for dashboard folder list filtering."""

from unittest.mock import MagicMock
from uuid import uuid4

import pytest
from sqlalchemy.orm.query import Query

from superset.dashboards.filters import DashboardFolderFilter
from superset.models.dashboard import Dashboard


@pytest.mark.parametrize("value", [None, ""])
def test_folder_filter_ignores_empty_values(value: str | None) -> None:
    query = MagicMock(spec=Query)

    result = DashboardFolderFilter("folder_id", MagicMock()).apply(query, value)

    assert result is query
    query.filter.assert_not_called()


def test_folder_filter_selects_uncategorized_dashboards() -> None:
    query = MagicMock(spec=Query)

    DashboardFolderFilter("folder_id", MagicMock()).apply(query, "uncategorized")

    condition = query.filter.call_args.args[0]
    assert condition.compare(Dashboard.folder_id.is_(None))


def test_folder_filter_selects_a_folder_by_uuid() -> None:
    query = MagicMock(spec=Query)
    folder_id = uuid4()

    DashboardFolderFilter("folder_id", MagicMock()).apply(query, folder_id)

    condition = query.filter.call_args.args[0]
    assert condition.compare(Dashboard.folder_id == folder_id)
